src/jace/optimization.py

```diff
@@ -0,0 +1,82 @@
+# JaCe - JAX Just-In-Time compilation using DaCe (Data Centric Parallel Programming)
+#
+# Copyright (c) 2024, ETH Zurich
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""JaCe specific optimizations.
+
+Currently just a dummy exists for the sake of providing a callable function.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Final, TypedDict
+
+from typing_extensions import Unpack
```

Comment:
Technically yes.
I think that the rule ignores the `typing` and `typing_extension` modules, because otherwise all import code would look like:
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import THE_REST
```

This was also a reason why I was against the TCH* rules, because in the end it is inconsistent, because anything you import from typing, except `TYPE_CHECKING` (for funny reasons), should be, by its very nature, inside this block!